features/git-rename-branch/destination_branch_exists_locally.feature

```diff
@@ -0,0 +1,30 @@
+Feature: git rename-branch: errors when the destination branch exists locally
+
+  As a developer trying to rename a branch with the name of an existing branch
+  I should see an error telling me that a branch with that name already exists
+  So that my new feature branch is unique.
```

Comment:
As a developer trying to rename a branch into an already existing branch
I want to the command to abort with an error message
So that I don't lose work by accidentally overwriting existing branches.